tests/postgres_tests/test_array.py

```diff
@@ -325,6 +339,33 @@ def test_usage_in_subquery(self):
             [self.objs[3]]
         )
 
+    def test_value_annotate(self):
+        value = models.Value([5], output_field=ArrayField(models.IntegerField()))
+        self.assertEqual(
+            NullableIntegerArrayModel.objects.annotate(value=value).first().value,
+            [5]
+        )
+
+    @unittest.expectedFailure
+    def test_value_annotate_empty(self):
+        # Ticket 28767
+        value = models.Value([], output_field=ArrayField(models.IntegerField()))
+        self.assertEqual(
+            NullableIntegerArrayModel.objects.annotate(value=value).first().value,
+            []
+        )
+
+    def test_value_annotate_aggregate(self):
```

Comment:
I see that only this test is failing with the change reverted. Additional test coverage is fine, but should be added in a separate PR or in a commit that precedes the bug fix.

Reply:
OK, I will split them out later